cypress/full/typeahead_page_spec.ts

```diff
@@ -0,0 +1,556 @@
+
+    it('clicking anywhere outside auto-fills "Option on blur" with the first option from the matches list', () => {
+      typeahead.clearInputAndSendKeys(onBlur, textToInput);
+      typeahead.isElementVisible(onBlur, typeahead.activeDropdown);
+      typeahead.clickOutside(`${onBlur} input`);
+      typeahead.isDemoContainsTxt(onBlur, stateForCheck);
+      typeahead.clearInputAndSendKeys(onBlur, anotherTextToInput);
+      typeahead.isElementVisible(onBlur, typeahead.activeDropdown);
+      typeahead.clickOutside(`${onBlur} input`);
+      typeahead.isDemoContainsTxt(onBlur, anotherStateForCheck);
+    });
+
+    it('If user clicks on any option then typeahead container auto-fills with a selected State', () => {
+      typeahead.isSelectedItemDisplayed(onBlur, stateForCheck);
+    });
+  });
+
+  describe('Append to body', () => {
+    const appendToBody = typeahead.exampleDemosArr.appendToBody;
+    const formTemplate = 'Model: ';
+    const stateForCheck = 'Rhode Island';
+    const anotherStateForCheck = 'Minnesota';
+    const textToInput = 'e o a';
+
+    it('User scrolls to Append to body sub-menu and sees typeahead input and typeahead card with "Model:"', () => {
+      typeahead.scrollToMenu('Append to body');
+      typeahead.isPreviewExist(appendToBody, formTemplate);
+      typeahead.isElementVisible(appendToBody, typeahead.inputSelector);
+    });
+
+    it('When a user starts to type then order of typed symbols and spaces between them are ignored', () => {
+      typeahead.clearInputAndSendKeys(appendToBody, textToInput);
+      typeahead.isElemTextCorrect(typeahead.activeDropdown, typeahead.dropdownItem, anotherStateForCheck, 1);
+      typeahead.isElemTextCorrect(typeahead.activeDropdown, typeahead.dropdownItem, stateForCheck, 2);
+    });
+
+    it('When user clicks on any item in typeahead drop-down, then typeahead container auto-fills with a selected State',
+      () => {
+        typeahead.isSelectedItemDisplayed(appendToBody, stateForCheck);
+      });
+  });
+
+  describe('No result', () => {
+    const noResult = typeahead.exampleDemosArr.noResult;
+    const formTemplate = 'Model: ';
+    const stateForCheck = 'Nevada';
+    const alertText = 'No Results Found';
+    const textToInput = 'qwerty';
+
+    it('User scrolls to No result sub-menu and sees typeahead input and typeahead card with "Model:"', () => {
+      typeahead.scrollToMenu('No result');
+      typeahead.isPreviewExist(noResult, formTemplate);
+      typeahead.isElementVisible(noResult, typeahead.inputSelector);
+    });
+
+    it('When user starts to type and if there are no matches found than an error message "No Results Found" is shown',
+      () => {
+        typeahead.clearInputAndSendKeys(noResult, textToInput);
+        typeahead.isDropdownDisabled(noResult, typeahead.activeDropdown);
+        typeahead.isElementVisible(noResult, typeahead.noResultAlert);
+        typeahead.isElemTextCorrect(noResult, typeahead.noResultAlert, alertText);
```

Comment:
isPreviewExist(noResult, 'No Results Found')

Reply:
can't use this method: it searches for `${baseSelector} .card-header` when i need to find `${baseSelector}.aleret`